tests/e2e-tests/specs/wp-admin/wp-admin-product-new.test.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * @format
+ */
+
+/**
+ * External dependencies
+ */
+import { activatePlugin } from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import { StoreOwnerFlow } from '../../utils/flows';
+import { clickTab, uiUnblocked } from '../../utils';
+
+const verifyPublishAndTrash = async () => {
```

Comment:
Just a question, should we perhaps move this to the StoreOwnerFlow file? I assume this can also be used for testing order and coupon publishing and trashing with a couple of tweaks so definitely reusable by other tests? Not a blocker but just thought I would ask.

Reply:
@kloon good call! I can definitely see it being reused for the coupon publishing. The new order, however, requires few more conditions to be verified once published (see below) but I will see how I can go around it (if possible):

```
// Verify
		await expect( page ).toMatchElement( '#message', { text: 'Order updated.' } );
		await expect( page ).toMatchElement( '#select2-order_status-container', { text: 'Processing' } );
		await expect( page ).toMatchElement(
			'#woocommerce-order-notes .note_content',
			{
				text: 'Order status changed from Pending payment to Processing.',
			}
		);
```

Since this PR is now approved, I will introduce the changes related to it in the next PR (new coupon and order tests).

Reply:
👍